lib/services/nodes-api-service.js

```diff
@@ -0,0 +1,96 @@
+// Copyright 2015, EMC, Inc.
+
+'use strict';
+
+var di = require('di');
+
+module.exports = nodeApiServiceFactory;
+di.annotate(nodeApiServiceFactory, new di.Provide('Http.Services.Api.Nodes'));
+di.annotate(nodeApiServiceFactory,
+    new di.Inject(
+        'Protocol.TaskGraphRunner',
+        'Services.Waterline',
+        'Errors',
+        'Logger',
+        '_',
+        'Promise'
+    )
+);
+function nodeApiServiceFactory(
+    taskGraphProtocol,
+    waterline,
+    Errors,
+    Logger,
+    _,
+    Promise
+) {
+    var logger = Logger.initialize(nodeApiServiceFactory);
+
+    function NodeApiService() {
```

Comment:
When I work on the bootstrap, I feel it is better if we can move the profile-api-service and template render service to the on-core, then some of the function (like rendering) can be shared in the job implementation (on-tasks).
Same to the nodes-api-service, in future we may have some internal logic to remove the nodes of a enclosure.

Of course, we can keep it here right now and move to on-core if we come across that situation.